### PR TITLE
Support for cable and infrared events.

### PR DESCRIPTION
**Pull request by pelikhan**
Refactored cable/infrared events to drop cogwheel. Fix for https://github.com/Microsoft/pxt-adafruit/issues/392
image

### CODE REVIEW

File: libs/cable/cable.ts [hunk 1]
```diff
@@ -42,14 +42,29 @@ namespace network {
     }
 
     /**
+     * Run some code when the cable receives data.
+     */
+    //% blockId=on_cable_received block="on cable received" blockGap=8
+    //% help=network/on-cable-received
+    //% parts="cable" group="Cable"
+    //% optionalVariableArgs
+    export function onCableReceived(handler: (num: number, nums?: number[], buf?: Buffer) => void) {
```
Comment (riknoll) [inline libs/cable/cable.ts]: Add `toolboxVariableArgs="1"`.

File: libs/infrared/ir.ts [hunk 2]
```diff
@@ -44,12 +44,27 @@ namespace network {
     /**
      * Run some code when the infrared receiver gets a packet.
      */
+    //% blockId=ir_on_infrared_received block="on infrared received" blockGap=8
+    //% help=network/on-infrared-received
+    //% parts="ir" group="Infrared"
+    //% optionalVariableArgs
+    export function onInfraredReceived(handler: (num: number, nums?: number[], buf?: Buffer) => void) {
```
Comment (riknoll) [inline libs/infrared/ir.ts]: Add `toolboxVariableArgs="1"`.

**Comment by abchatra** [comment]
I am confused. I thought we removed cable api as it is kind of hard for kids to use it and there was no documentation. Why does this come back with this change?

**Comment by pelikhan** [comment]
It's gone, I added the package for testing.

**Comment by abchatra** [comment]
Got it

**Pull request merged**